tidy3d/plugins/smatrix/ports/lumped.py

```diff
@@ -0,0 +1,173 @@
+"""Class and custom data array for representing a scattering matrix port based on lumped circuit elements."""
+import pydantic.v1 as pd
+import numpy as np
+from typing import Union
+
+from ....constants import OHM
+from ....components.geometry.base import Box
+from ....components.types import Complex, FreqArray, Axis
+from ....components.base import cached_property
+from ....components.lumped_element import LumpedResistor
+from ....components.monitor import FieldMonitor
+from ....components.source import UniformCurrentSource, GaussianPulse
+from ....components.validators import assert_plane
+from ....components.data.data_array import DataArray
+from ....exceptions import ValidationError
+
+DEFAULT_PORT_NUM_CELLS = 3
+
+
+class LumpedPortDataArray(DataArray):
+    """Port parameter matrix elements for lumped ports.
+
+    Example
+    -------
+    >>> port_in = ['port1', 'port2']
+    >>> port_out = ['port1', 'port2']
+    >>> f = [2e14]
+    >>> coords = dict(
+    ...     port_in=ports_in,
+    ...     port_out=ports_out,
+    ...     f=f
+    ... )
+    >>> fd = LumpedPortDataArray((1 + 1j) * np.random.random((2, 2, 1)), coords=coords)
+    """
+
+    __slots__ = ()
+    _dims = ("port_out", "port_in", "f")
+    _data_attrs = {"long_name": "lumped port matrix element"}
+
+
+class LumpedPort(Box):
+    """Class representing a single lumped port"""
+
+    name: str = pd.Field(
+        ...,
+        title="Name",
+        description="Unique name for the port.",
+        min_length=1,
+    )
+
+    voltage_axis: Axis = pd.Field(
+        ...,
+        title="Voltage Integration Axis",
+        description="Specifies the axis along which the E-field line integral is performed when "
+        "computing the port voltage. The integration axis must lie in the plane of the port.",
+    )
+
+    impedance: Complex = pd.Field(
+        50,
+        title="Reference impedance",
+        description="Reference port impedance for scattering parameter computation.",
+        units=OHM,
+    )
+
+    num_grid_cells: Union[pd.PositiveInt, None] = pd.Field(
```

Comment:
I think you can do `Optional[pd.PostiveInt]` where `Optional` is imported from `typing`.